src/main/anovos/data_ingest/data_ingest.py

```diff
@@ -41,13 +55,129 @@ def read_dataset(spark, file_path, file_type, file_configs={}):
         e.g. {"header": "True","delimiter": "|","inferSchema": "True"} for csv files.
         All the key/value pairs in this argument are passed as options to DataFrameReader,
         which is created using SparkSession.read. (Default value = {})
+    parquet_conversion
+        This boolean flag provides an option to whether convert the file type to parquet or not.
+        Schema treatment is highly recommended when using this method, as the schema when converted will
+        be kept the same as the original file, thus might defeat the purpose of converting to parquet.
+        (Default value = False)
+    parquet_output_path
+        This argument is passed as output path to write out parquet file, if parquet_conversion
+        is set to True. (Default value = '')
+    treatment
+        This boolean flag provides an option to whether treat the dataframe schema or not. It might not
+        work on parquet filetype, as parquet has its own strict Schema.
+        (Default value = False)
+    id_cols
+        This argument is a list contains all the identifier columns, such as ID, Number, Code, etc
+        when treatment is set to True.
+        (Default value = [])
+    threshold_num
+        This argument determines the numerical threshold of cardinality for every column. If the number of
+        unique values of a column is larger than threshold_num, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 50)
+    threshold_ratio
+        This argument determines the cardinality ratio for every column. The cardinality ratio is defined
+        by number of non-null unique values divided by number of non-null total values in a column.
+        If the cardinality ratio is larger than threshold_ratio, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 0.005)
+    threshold_string
+        This argument determines the string ratio for every column. The string ratio is defined
+        by number of non-null string values divided by number of non-null total values in a column.
+        If the string ratio is smaller than threshold_ratio, its DataType will be treated as Double.
+        Otherwise its DataType will be treated as String. (Default value = 0.5)
 
     Returns
     -------
     DataFrame
 
     """
     odf = spark.read.format(file_type).options(**file_configs).load(file_path)
+    if parquet_conversion:
+        if not parquet_output_path:
+            raise TypeError(
+                "parquet_output_path cannot be blank for parquet conversion"
+            )
+    if treatment:
+        if id_cols:
+            for col in id_cols:
+                if col not in odf.columns:
+                    raise TypeError("Invalid input for id_cols: " + col)
```

Comment:
This is not a `TypeError` but a `ValueError`. You should also mention explicitly mention the problem: The column is not in the dataframe. ("Invalid" can mean a lot of things.)

Reply:
Resolved with https://github.com/anovos/anovos/pull/268/commits/b3044847cfb728a4b4bec2d105ea7253e325643e